Added reviews

ronits-wework opened the pull request:
Added reviews
Calculated restaurant rating according to reviews
Added a Add Review button to Restaurant view
Added tests for restaurant rating

ronits-wework committed:
Add reviews

ronits-wework committed:
Calculate restaurant rating according to reviews

ronits-wework committed:
Bind rating updates to after_X hooks

ronits-wework committed:
Add a Add Review button to Restaurant view
Increase precision of restaurant rating
Add a test for restaurant rating average
Add DatabaseCleaner to clean the DB after tests run

limormek commented on config/application.rb:
```diff
@@ -7,6 +7,11 @@
+
+  MIN_REVIEW_RATING = 0.0
+  MAX_REVIEW_RATING = 3.0
+  DELIVERY_TIMES = 15.step(120, 15).to_a.freeze
```
nice!

limormek commented on app/models/review.rb:
```diff
@@ -0,0 +1,55 @@
+                                     less_than_or_equal_to: WeEat::MAX_REVIEW_RATING }
+
+  def add_rating_to_restaurant!
+    review = self
```
no need in this assignment, you can later call restaurant without any indication, and it's like calling self.restaurant.

limormek commented on app/models/review.rb:
```diff
@@ -0,0 +1,55 @@
+  end
+
+  def update_rating_to_restaurant!
+    review = self
```
same here

limormek commented on app/models/review.rb:
```diff
@@ -0,0 +1,55 @@
+  end
+
+  def delete_rating_from_restaurant!
+    review = self
```
and here

limormek commented on spec/models/review_spec.rb:
```diff
@@ -0,0 +1,19 @@
+RSpec.describe Review, type: :model do
+
+  describe "test average rating calculation" do
+    #let!(:review_instance) { FactoryGirl.create(:review) }
```
minor: can delete this line

ronits-wework committed:
Add a Add Review button to Restaurant view
Increase precision of restaurant rating
Add a test for restaurant rating average
Add DatabaseCleaner to clean the DB after tests run

ronits-wework committed:
Added task to import csv from 10bis

ronits-wework committed:
Added additional fields from csv to Restaurant

limormek commented on app/models/restaurant.rb:
```diff
@@ -16,25 +16,19 @@ class Restaurant < ApplicationRecord
-  # end
+  def set_rating_in_bounds!
+    if self.rating != nil
+      self.rating = [[self.rating, WeEat::MAX_REVIEW_RATING].min, WeEat::MIN_REVIEW_RATING].max
```
minor: you could change to something like (and no need of 'self':
`rating = rating ? [[self.rating, WeEat::MAX_REVIEW_RATING].min, WeEat::MIN_REVIEW_RATING].max : nil`

ronits-wework committed:
fixed PR comments

ronits-wework committed:
Fixed rake task - import restaurants from CSV
